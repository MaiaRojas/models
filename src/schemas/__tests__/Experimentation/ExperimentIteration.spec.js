const mongoose = require('mongoose/browser');
const { ExperimentIterationSchema } = require('../../')(mongoose);

describe('ExperimentIterationSchema', () => {
  it('should fail validation when fields missing', () => {
    const doc = new mongoose.Document({}, ExperimentIterationSchema);
    expect(doc.validateSync().errors).toMatchSnapshot();
  });

  it('should validate document with valid data whith minimal data', () => {
    const doc = new mongoose.Document({
      version: '2',
    }, ExperimentIterationSchema);
    return doc.validate();
  });
});
