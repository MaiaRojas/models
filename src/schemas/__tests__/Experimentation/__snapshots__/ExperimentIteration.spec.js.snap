// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperimentIterationSchema should fail validation when fields missing 1`] = `
Object {
  "version": [ValidatorError: Path \`version\` is required.],
}
`;
